src/main/java/frc/robot/Robot.java

```diff
@@ -112,13 +119,31 @@ public void autonomousRoutine() {
   @Override
   public void teleopSetup() {
     // TODO Set up bindings
+    this.driverPad.setMode(ControllerBindings.LEFT_STICK, ControllerBindings.Y_AXIS, new SquaredJoystickMode());
+    this.driverPad.setMode(ControllerBindings.RIGHT_STICK, ControllerBindings.X_AXIS, new SquaredJoystickMode());
+
+    
+    this.driverPad.bind(ControllerBindings.INTAKE_EXTEND, PressType.PRESS, () -> this.ballSubsystem.extendOuter());
+    this.driverPad.bind(ControllerBindings.INTAKE_RETRACT, PressType.PRESS, () -> this.ballSubsystem.retractOuter());
+    this.driverPad.bind(ControllerBindings.INTAKE_REVERSE, PressType.PRESS, () -> this.ballSubsystem.reverse());
+
+    this.driverPad.bind(ControllerBindings.SPIN_UP, PressType.PRESS, () -> this.ballSubsystem.readyToShoot());
+    this.driverPad.bind(ControllerBindings.SHOOT, PressType.PRESS, () -> this.ballSubsystem.startShooter());
+    this.driverPad.bind(ControllerBindings.SHOOT, PressType.RELEASE, () -> this.ballSubsystem.stopShooter());
+
+
+    this.driverPad.bind(ControllerBindings.CLIMBER_EXTEND, PressType.PRESS, () -> this.climberSubsystem.extendOuterClimber());
+    this.driverPad.bind(ControllerBindings.CLIMBER_RETRACT, PressType.PRESS, () -> this.climberSubsystem.retractOuterClimber());
+    this.driverPad.bind(ControllerBindings.CLIMBER_ROTATE, PressType.PRESS, () -> this.climberSubsystem.angleOuter(0.0));
```

Comment:
Gamepad bindings and mappings won't work until the `driverPad` is `start()`ed. You only want it to run during teleop though, and it should be `cancel()`led during other lifecycle phases.

The easiest way to do this is simply using `registerLifecycleComponent`, like so:

https://github.com/FRC3161/WoodieFlowers2020/blob/66b24e53f63d60d724c5860ebefe921741cde78c/src/main/java/frc/robot/Robot.java#L87

The implementation behind that is here, for your understanding:

https://github.com/FRC3161/Yggdrasil/blob/f367d6bd1adb20eeac216a1fa5ff877efae19bc2/src/main/java/ca/team3161/lib/utils/controls/AbstractController.java#L161

Reply:
Implemented this with the driverpad, ballsubsystem, and drivetrain in the recent commit.